src/main/java/org/shanerx/tradeshop/commands/commandrunners/GUICommand.java

```diff
@@ -0,0 +1,293 @@
+/*
+ *
+ *                         Copyright (c) 2016-2019
+ *                SparklingComet @ http://shanerx.org
+ *               KillerOfPie @ http://killerofpie.github.io
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *                http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  NOTICE: All modifications made by others to the source code belong
+ *  to the respective contributor. No contributor should be held liable for
+ *  any damages of any kind, whether be material or moral, which were
+ *  caused by their contribution(s) to the project. See the full License for more information.
+ *
+ */
+
+package org.shanerx.tradeshop.commands.commandrunners;
+
+import de.themoep.inventorygui.GuiElement;
+import de.themoep.inventorygui.GuiElementGroup;
+import de.themoep.inventorygui.GuiPageElement;
+import de.themoep.inventorygui.GuiStateElement;
+import de.themoep.inventorygui.InventoryGui;
+import de.themoep.inventorygui.StaticGuiElement;
+import org.bukkit.Material;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.meta.BookMeta;
+import org.bukkit.inventory.meta.Damageable;
+import org.bukkit.inventory.meta.FireworkMeta;
+import org.bukkit.inventory.meta.ItemMeta;
+import org.shanerx.tradeshop.TradeShop;
+import org.shanerx.tradeshop.commands.CommandPass;
+import org.shanerx.tradeshop.item.ShopItemSide;
+import org.shanerx.tradeshop.item.ShopItemStack;
+import org.shanerx.tradeshop.item.ShopItemStackSettingKeys;
+import org.shanerx.tradeshop.shop.Shop;
+import org.shanerx.tradeshop.utils.objects.ObjectHolder;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation of CommandRunner for plugin commands that generate GUI screens
+ *
+ * @since 2.6.0
+ */
+public class GUICommand extends CommandRunner {
+
+    //region Util Variables
+    //------------------------------------------------------------------------------------------------------------------
+
+
+    protected final String[] MENU_LAYOUT = {"ad bc"},
+            EDIT_LAYOUT = {"aggggggga", "ap c s na"},
+            ITEM_LAYOUT = {"u ggggggg", "j hhhhhhh", "ap cbs na"},
+            SETTING_LAYOUT = {"ggggggggg", "hhhhhhhhh", "ap cbs na"},
+            WHAT_MENU = {"141125333", "1qqq2eee3", "11p123n33"};
+    protected List<ShopItemStack> costItems = new ArrayList<>(),
+            productItems = new ArrayList<>();
+    protected List<Boolean> costItemsRemoval = new ArrayList<>(),
+            productItemsRemoval = new ArrayList<>();
```

Comment:
Idea doesn't seem to care about it being moved but since the GUI stuff is so finicky, I want to save this for a release that will have more thorough testing. I also want time to see if I can write the GUI code cleaner.